File: qc-core/src/main/java/com/eager/questioncloud/core/domain/point/infrastructure/repository/UserPointRepository.java
package com.eager.questioncloud.core.domain.point.infrastructure.repository;

import com.eager.questioncloud.core.domain.point.model.UserPoint;
import org.springframework.stereotype.Repository;

@Repository
public interface UserPointRepository {
    UserPoint getUserPoint(Long userId);

    Boolean usePoint(Long userId, int amount);

    void chargePoint(Long userId, int amount);

    UserPoint save(UserPoint userPoint);
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/point/infrastructure/repository/UserPointRepository.kt
package com.eager.questioncloud.core.domain.point.infrastructure.repository

import com.eager.questioncloud.core.domain.point.model.UserPoint
import org.springframework.stereotype.Repository

@Repository
interface UserPointRepository {
    fun getUserPoint(userId: Long): UserPoint

    fun usePoint(userId: Long, amount: Int): Boolean

    fun chargePoint(userId: Long, amount: Int)

    fun save(userPoint: UserPoint): UserPoint

    fun deleteAllInBatch()
}
